### DISCUSSION TITLE
Why there is no any "Reset all recording settings to their defaults" button at all?

### DISCUSSION THREAD

**Discussion by ghost** [obsproject/obs-studio#7864] [Development Questions]
I watched a YouTube video and the person who recorded himself says that I **must** Show Settings Folder, close or exit OBS, empty the settings folder and then relaunch OBS but why I have to do all this to just reset all settings to their defaults?

Why there is no button that does just that?

The person that speaks in the YouTube video also said the consequences of doing this like losing more than custom settings.

Why there is no a "Reset all settings to their defaults" button that does just that?

Here is the YouTube video if you want to watch by yourself [https://www.youtube.com/watch?v=AJ2SjuHRfBo](url)

**Comment by derrod** [comment]
Because users would press it, ignore the warnings, and then complain to us.

It's also never really necessary, if you believe there's an issue with your profile or scene collection just create a blank new one. The global settings can be reset reasonably easily by removing the `global.ini` file from the settings folder.

(And also because it's technically somewhat difficult to implement since you can't just indiscriminately delete everything unless you also want to remove your scene collections or certain other data that may not need to be reset.)

**Reply by WizardCM** [reply]
Seconded - don't want an easy footgun, and it hasn't really been requested.

**Reply by ghost** [reply]
> Because users would press it, ignore the warnings, and then complain to us.

And isn't this a complaint?

Also these users are idiots and stupid and this is their faults if they ignored the warnings. You should **NOT** take any responsibilities and actions for their nuts and mistakes.

There should be an **automatic bot** that automatically asks the user that complains if he or she read or notice the warning message.

If he or she says a negative answer then you should not do anything. The bot should just tell him or her that he or she should **not** ignore the warnings next time and that's it. The issue is **automatically** closed by the automatic bot.

Another possible solution to this problem is to add a countdown of at least 30 seconds with a text view that explicitly says in caps lock:

> YOU MUST READ THE ENTIRE WARNING MESSAGE BEFORE CONTINUING. OTHERWISE DO NOT COMPLAIN TO US IF SOMETHING BAD OR TERRIBLE HAPPENED TO YOU AND YOU HAVE BEEN WARNED THAT THIS MAY OR MIGHT HAPPEN. WE WILL NOT TAKE ANY RESPONSIBILITIES AND ACTIONS FOR USERS THAT IGNORED AND DID NOT READ ALL WARNING MESSAGES AT ALL.

For each warning message and don't let the user to continue until the 30 seconds have been elapsed **and** ticked the "Acknowledged" checkbox.

I personally don't think that this solution is better than the automatic bot solution suggested above but this may cause the user to read the warning message and not ignore it.

> (And also because it's technically somewhat difficult to implement since you can't just indiscriminately delete everything unless you also want to remove your scene collections or certain other data that may not need to be reset.)

I know nothing about the complexity and difficulty of implementing this "Reset all settings to their defaults" feature to work accurately and precisely.

I do know that if **all** custom settings are saved in a single `settings.ini` or `config.cfg` file then just deleting this single `settings.ini` or `config.cfg` file or moving it to the recycle bin and let OBS create a new fresh `settings.ini` or `config.cfg` to reset all settings to their defaults is as simple as that.

But if you say that this is not so trivial then so be it.

**Reply by Fenrirthviti** [reply]
As we already mentioned, you can easily perform a full "reset" of any meaningful setting in OBS by creating both a new profile, and new scene collection from their respective menus. Having a global reset button that can be accidentally pressed and wipe a whole setup to save a few clicks is a bad idea.

**Reply by ghost** [reply]
> As we already mentioned, you can easily perform a full "reset" of any meaningful setting in OBS by creating both a new profile, and new scene collection from their respective menus. Having a global reset button that can be accidentally pressed and wipe a whole setup to save a few clicks is a bad idea.

I did **NOT** expect a **global** reset button that resets and removes everything but only a button that resets all settings **about** how your input and output devices (screen, monitor, LED, speakers, headsets, headphones, microphone and etc) are recorded and how the output file is encoded.

The thing is this that I messed up with the settings that deal with how both video and audio are recorded and the output video file was bad both mkv and mp4.

Before I messed up with these settings the output video file was good both mkv and mp4.

So I just wanted to reset **only** these settings but I didn't find any button that does just that.

But of course that this is really **bad**, even a **terrible**, idea to have such a global reset button in the GUI that both resets, removes and deletes everything on click.

Perhaps I had to ask

> Why there is no a "Reset all recording settings to their defaults" button at all

In first place to be more accurate, exact and clear.

Alright I have corrected my question to be more accurate, exact and clear.

Thank you for your replies and the correction.
